jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt

```diff
@@ -71,6 +83,26 @@ abstract class CodeWhispererCodeCoverageTracker(
         addAndGetTotalTokens(event.newLength - event.oldLength)
     }
 
+    internal fun extractRangeMarkerString(rangeMarker: RangeMarker?): String? = runReadAction {
+        rangeMarker?.range?.let { myRange -> rangeMarker.document.getText(myRange) }
+    }
+
+    internal fun getAcceptedTokensDelta(originalRecommendation: String, modifiedRecommendation: String): Int {
+        val editDistance = getEditDistance(modifiedRecommendation, originalRecommendation).toInt()
+        val originalRecommendationLength = originalRecommendation.length
+        val modifiedRecommendationLength = modifiedRecommendation.length
+        return when {
+            originalRecommendationLength < modifiedRecommendationLength ->
+                (modifiedRecommendationLength - editDistance).coerceAtMost(originalRecommendationLength)
+
+            originalRecommendationLength > modifiedRecommendationLength -> originalRecommendationLength - editDistance
+
+            else -> originalRecommendationLength - editDistance
+        }
```

Comment:
For the whole logic, if you want: `return max(original, modified) - editDistance`

Reply:
yea good point, that's right.